### PR TITLE
chore: add Claude Code AI assistant configuration

### PR DESCRIPTION
**Pull request by ahmedAlaaInstabug**
## Description of the change

Adds Claude Code AI assistant configuration to the repository. This includes project context documentation, Flutter/Dart coding rules, 12 slash commands for common workflows, and 3 custom agents for codebase exploration.

### Slash Commands

| Command | Description | Example |
|---|---|---|
| `/create-pr` | Automates commit, PR template filling, and draft PR creation via GitHub CLI | `/create-pr taskUrl=https://instabug.atlassian.net/browse/MOB-20541` |
| `/describe-pr` | Generates a comprehensive PR description from the current branch changes using the repo's PR template | `/describe-pr` (auto-detects PR from current branch) |
| `/apply-pr-reviews` | Fetches unresolved PR review comments and applies fixes one by one with user confirmation | `/apply-pr-reviews 123` or `/apply-pr-reviews --auto` or `/apply-pr-reviews --dry-run` |
| `/pr-review` | Runs comprehensive code review across 6 dimensions: logic errors, Dart standards, platform channel safety, test coverage, performance, and SDK best practices | `/pr-review` (reviews current branch diff against master) |
| `/review` | Runs a checklist-based code review covering platform channels, public API compatibility, testing, error handling, performance, security, and monorepo impact | `/review` (reviews current changes) |
| `/create-rca` | Generates a Root Cause Analysis document by gathering PR diffs and Jira ticket details, then filling an RCA template | `/create-rca taskUrl=https://instabug.atlassian.net/browse/MOB-21429 prUrls=https://github.com/Instabug/Luciq-Flutter/pull/45` |
| `/create-jira-workitem` | Creates a Jira work item prefixed with `[Flutter]` in the MOB project, auto-assigned to you | `/create-jira-workitem Add network timeout configuration to Dio interceptor` |
| `/create-public-release-notes` | Transforms internal wiki release notes into public-facing release notes for SDK consumers, stripping ticket IDs and internal jargon | `/create-public-release-notes releaseNotesUrl=https://your-wiki.com/Flutter+19.3.0+Internal+Release+Notes` |
| `/new-feature` | Scaffolds a new SDK feature module: Pigeon API, module class, models, exports, and tests | `/new-feature name=Surveys` |
| `/new-api` | Adds a new public API method to an existing feature module through the full Pigeon pipeline | `/new-api module=BugReporting method=setCustomBrandingImage` |
| `/debug-sdk` | Troubleshoots common SDK issues: initialization, crash capture, network logging, screen loading, widget rendering, and build failures | `/debug-sdk SDK not capturing network requests from Dio` |
| `/research-codebase` | Researches and documents how a specific part of the codebase works by tracing flows from public API to platform channels | `/research-codebase topic=how screen loading tracking works` |

### Custom Agents

| Agent | Description |
|---|---|
| `codebase-analyzer` | Analyzes HOW code works - traces flows, understands patterns and implementation details |
| `codebase-locator` | Finds WHERE code lives in the monorepo - locates files, classes, and functions |
| `codebase-pattern-finder` | Finds existing code patterns and examples to follow when implementing new features |

### Other Files

- **`CLAUDE.md`** - Project overview, monorepo structure, common commands, architecture docs, and output/style guidelines
- **`rules.md`** - Flutter/Dart best practices covering code quality, state management, testing, theming, and accessibility
- **`.claude/settings.local.json`** - Permission allowlist for safe read-only commands

## Type of change
- [x] New feature (non-breaking change that adds functionality)

### Development
- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
### Code review
- [x] This pull request has a descriptive title and information useful to a reviewer

### CODE REVIEW

**Review by andrewdanieldev — COMMENTED** [review]
## Code Review

### Overview
This PR adds a complete Claude Code AI assistant setup: `CLAUDE.md`, 12 slash commands, 3 custom agents, Flutter/Dart rules, and a local settings file. The core setup is solid - well-structured commands with clear parameters, good agent scoping, and practical project context.

### What's Good
- Well-structured commands with clear parameters, steps, and error handling
- The three specialized agents (analyzer, locator, pattern-finder) are well-scoped with clear boundaries
- `CLAUDE.md` provides solid project context (monorepo structure, Pigeon flow, module pattern)
- The `apply-pr-reviews` command has a thoughtful workflow with user confirmation at each step

### Summary
Main concerns: (1) committing `settings.local.json` with write permissions, (2) the oversized generic Flutter rules file inflating context, and (3) inconsistent CLI tool references across commands. See inline comments for details.

File: .claude/settings.local.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
```
Comment (andrewdanieldev) [inline .claude/settings.local.json]: **Critical:** `settings.local.json` is meant for per-developer local configuration. Committing it forces everyone to use the same permission allowlist and prevents individual customization.

This should either be:
- Renamed to `settings.json` (the shared project-level config), or
- Added to `.gitignore` and documented in CLAUDE.md as a recommended setup step

File: .claude/settings.local.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      "Bash(file:*)",
+      "Bash(diff:*)",
+      "Bash(grep:*)",
+      "Bash(rg:*)",
+      "Bash(tr:*)",
+      "Bash(cut:*)",
+      "Bash(jq:*)",
+      "Bash(gh --version)",
+      "Bash(gh auth:*)",
+      "Bash(git add:*)",
```
Comment (andrewdanieldev) [inline .claude/settings.local.json]: **Warning:** Auto-allowing `git add:*` and `git commit:*` means Claude can stage and commit changes without user confirmation. This is risky for a shared config pushed to all developers. Consider removing these two entries, or at minimum documenting them as opt-in that developers can add to their own local settings.

File: .claude/commands/create-pr.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+- Add route matching for automatic screen detection
+- Include unit tests for edge cases
+```
+
+**Present the suggested commit message and wait for user approval.**
+
+Once approved, stage and commit:
+
+```bash
+git add -A
```
Comment (andrewdanieldev) [inline .claude/commands/create-pr.md]: **Warning:** `git add -A` stages everything including potentially untracked files, `.env` files, or build artifacts. Consider using `git add <specific-files>` or at minimum showing the user what will be staged (`git status`) and asking for confirmation before running `git add -A`.

File: .claude/commands/create-pr.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+**Present the title to the user for approval.**
+
+Once approved:
+
+```bash
+# Push the branch
+git push -u origin HEAD
+
+# Save PR description
+cat > /tmp/pr-description.txt << 'EOF'
```
Comment (andrewdanieldev) [inline .claude/commands/create-pr.md]: **Suggestion:** Using a fixed path `/tmp/pr-description.txt` is fragile on shared systems and could collide if two people run this simultaneously. Consider using `mktemp` instead:
```bash
tmpfile=$(mktemp)
cat > "$tmpfile" << 'EOF'
...
EOF
gh pr create ... --body-file "$tmpfile"
rm "$tmpfile"
```

File: .claude/commands/create-rca.md [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+  - Example: `taskUrl=https://instabug.atlassian.net/browse/MOB-21429`
+
+## 0. Prerequisites Check
+
+**IMPORTANT**: This command requires GitHub CLI (`gh`) and Jira CLI (`jira`). Check if they are installed:
+
+```bash
+gh --version 2>/dev/null || echo "GH_CLI_NOT_INSTALLED"
+jira version 2>/dev/null || echo "JIRA_CLI_NOT_INSTALLED"
+```
```
Comment (andrewdanieldev) [inline .claude/commands/create-rca.md:23]: **Warning:** This references the `jira` CLI (`ankitpokhrel/jira-cli`), which is a non-standard third-party tool, while `create-jira-workitem.md` references `acli` - a different Jira CLI. Pick one tool for consistency across all commands.

Also consider whether the Atlassian MCP tools (already available in the tool list) could replace the CLI dependency entirely, removing the install/auth prerequisite.

File: .claude/commands/create-jira-workitem.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+---
+
+Create a Jira work item using Atlassian CLI.
+
+Example:
+
+```
+acli jira workitem create \
+    --summary "Task Title" \
+    --description "Task description in Atlassian Document Format" \
```
Comment (andrewdanieldev) [inline .claude/commands/create-jira-workitem.md:12]: **Warning:** This uses `acli` while `create-rca.md` uses `jira` CLI (`ankitpokhrel/jira-cli`) - two different Jira CLI tools across commands. Should be consistent.

File: .claude/rules/flutter-dart.md [hunk 7]
```diff
@@ -0,0 +1,777 @@
+# AI rules for Flutter
```
Comment (andrewdanieldev) [inline .claude/rules/flutter-dart.md]: **Warning:** At 777 lines, most of this is generic Flutter/Dart best practice (theming, fonts, color schemes, accessibility, layout, routing with GoRouter, etc.) that isn't specific to this SDK project. Since rules files are loaded into every conversation, this bloats the context window unnecessarily.

Recommend trimming to only project-specific conventions (module pattern, Pigeon usage, testing patterns with Mockito, naming conventions). Generic Flutter knowledge is already part of Claude's training data and doesn't need to be re-injected every session.

File: .claude/commands/describe-pr.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+Generate a PR description using the repository's `.github/pull_request_template.md`.
+
+## Steps
+
+1. **Read the PR template:**
+   - Read `.github/pull_request_template.md`
+   - If not found, inform the user and wait for instructions
+
+2. **Identify the PR:**
+   - Check if the current branch has an associated PR: `gh pr view --json url,number,title,state 2>/dev/null`
```
Comment (andrewdanieldev) [inline .claude/commands/describe-pr.md:16]: **Warning:** If no PR exists, this silently creates one with `gh pr create --fill`. This should confirm with the user first before creating a PR - it's a visible action on a shared system.

File: .claude/commands/apply-pr-reviews.md [hunk 9]
```diff
@@ -0,0 +1,229 @@
...
+Then fetch the PR associated with this branch using GitHub CLI:
+```bash
+gh pr list --head <branch-name> --json number,title,url,state --limit 1
+```
+
+## 2. Fetch Unresolved PR Review Comments
+
+Use GitHub CLI to fetch inline review comments (file-specific comments):
+```bash
+gh api repos/{owner}/{repo}/pulls/{pr_number}/comments --jq '.[] | select(.in_reply_to_id == null) | {id: .id, path: .path, line: .line, body: .body, user: .user.login, created_at: .created_at}'
```
Comment (andrewdanieldev) [inline .claude/commands/apply-pr-reviews.md:63]: **Suggestion:** The `select(.in_reply_to_id == null)` filter only removes replies, not resolved threads. The GitHub API doesn't expose resolution status on individual review comments via this endpoint. Consider noting this limitation, or using `gh api graphql` with the `PullRequestReviewThread.isResolved` field to actually filter out resolved threads.

File: CLAUDE.md [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+- Tests live in `packages/luciq_flutter/test/` (31+ test files)
+- Uses **Mockito** for mocking — generated `.mocks.dart` files via `build_runner`
+- After adding `@GenerateMocks` annotations, run `melos generate --no-select`
+- E2E tests are in `/e2e/` (C# + Appium, separate from Flutter unit tests)
+
+## Code Quality
+
+- Linting: `analysis_options.yaml` extends `package:lint/analysis_options_package.yaml`
+- Generated files (`*.g.dart`, `*.mocks.dart`) are excluded from analysis
+- CI: CircleCI with Flutter 3.24.3
```
Comment (andrewdanieldev) [inline CLAUDE.md]: **Suggestion:** "Flutter 3.24.3" will go stale. Consider removing the version number or referencing the CI config file instead (e.g., "see `.circleci/config.yml` for the current Flutter version").

File: CLAUDE.md [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+- Apply them silently. Do not re-announce learned behavior.
+- If the user corrects a mistake: fix it, remember it, move on.
+
+## Scope Control
+- Do not add features beyond what was asked.
+- Do not refactor surrounding code when fixing a bug.
+- Do not create new files unless strictly necessary.
+
+## Override Rule
+User instructions always override this file.
```
Comment (andrewdanieldev) [inline CLAUDE.md]: **Nit:** Missing newline at end of file.

File: CLAUDE.md [hunk 12]
```diff
@@ -0,0 +1,154 @@
+# CLAUDE.md
+
+This file provides guidance to Claude Code (claude.ai/code) when working with code in this repository.
+
+## Project Overview
+
+Luciq Flutter SDK — the official Flutter plugin for Luciq, an Agentic Observability Platform for Mobile. Provides bug reporting, crash reporting, APM, surveys, in-app replies, feature requests, and session replay. Current version: 19.2.2.
```
Comment (andrewdanieldev) [inline CLAUDE.md:7]: **Suggestion:** The naming is inconsistent across the PR. This file uses "Luciq" throughout, but several commands reference "Instabug" URLs (e.g., `create-rca.md` has `https://github.com/Instabug/Luciq-Flutter/pull/45`, `create-pr.md` has `https://instabug.atlassian.net`). Should be consistent - either mask the internal org name in examples or use the same name everywhere.

File: .claude/settings.local.json (hunk 1 above)
Comment (ahmedAlaaInstabug) [reply .claude/settings.local.json]: Applied in efee99a. Added `.claude/settings.local.json` to `.gitignore` and untracked the existing file - each developer now keeps their own local permission allowlist.

File: .claude/settings.local.json (hunk 2 above)
Comment (ahmedAlaaInstabug) [reply .claude/settings.local.json]: Applied in efee99a. Removed both `Bash(git add:*)` and `Bash(git commit:*)` entries. The file is also now gitignored so no one picks them up by default.

File: .claude/commands/create-pr.md (hunk 3 above)
Comment (ahmedAlaaInstabug) [reply .claude/commands/create-pr.md]: Applied in efee99a. Replaced `git add -A` with a `git status` review step followed by explicit file staging (`git add <file1> <file2>`). The workflow now asks for user approval of the file list before committing.

File: .claude/commands/create-pr.md (hunk 4 above)
Comment (ahmedAlaaInstabug) [reply .claude/commands/create-pr.md]: Applied in efee99a. Switched to `mktemp` to avoid collisions on shared systems.

File: .claude/commands/create-rca.md (hunk 5 above)
Comment (ahmedAlaaInstabug) [reply .claude/commands/create-rca.md:23]: Applied in efee99a. Unified both commands on `acli` (Atlassian CLI) so there is a single Jira tool across the repo. Kept CLI-based rather than MCP for now to match `create-jira-workitem.md`; we can migrate to the Atlassian MCP tools later as a follow-up if we want to drop the CLI dependency entirely.

File: .claude/commands/create-jira-workitem.md (hunk 6 above)
Comment (ahmedAlaaInstabug) [reply .claude/commands/create-jira-workitem.md:12]: Applied in efee99a. `create-rca.md` now uses `acli` to match this file, so both Jira commands share the same tool.

File: .claude/rules/flutter-dart.md (hunk 7 above)
Comment (ahmedAlaaInstabug) [reply .claude/rules/flutter-dart.md]: Applied in efee99a. Trimmed the file from 777 to 88 lines - removed generic Flutter/Dart content (theming, colors, fonts, layout, routing, accessibility, state management, go_router, etc.) and kept only project-specific rules: module pattern, Pigeon usage, public API barrel, Mockito testing, code generation, and melos commands.

File: .claude/commands/describe-pr.md (hunk 8 above)
Comment (ahmedAlaaInstabug) [reply .claude/commands/describe-pr.md:16]: Not applied in this PR. I tried adding an explicit confirmation step, but reverted - we prefer the current behavior of `gh pr create --fill` as the path of least friction for this command. Happy to revisit if we see accidental PR creations in practice.

File: .claude/commands/apply-pr-reviews.md (hunk 9 above)
Comment (ahmedAlaaInstabug) [reply .claude/commands/apply-pr-reviews.md:63]: Applied in efee99a. Added a note explaining that `in_reply_to_id == null` only filters replies, not resolved threads, and included the `gh api graphql` snippet using `PullRequestReviewThread.isResolved` as the proper way to exclude resolved threads.

File: CLAUDE.md (hunk 10 above)
Comment (ahmedAlaaInstabug) [reply CLAUDE.md]: Applied in efee99a. Replaced the hardcoded version with a pointer to `.circleci/config.yml` as the source of truth.

File: CLAUDE.md (hunk 11 above)
Comment (ahmedAlaaInstabug) [reply CLAUDE.md]: Applied in efee99a. Added the trailing newline.

File: CLAUDE.md (hunk 12 above)
Comment (ahmedAlaaInstabug) [reply CLAUDE.md:7]: Applied in efee99a. Fixed `github.com/Instabug/Luciq-Flutter` -> `github.com/luciqai/luciq-flutter-sdk` in `create-rca.md` examples (the repo path was wrong). Kept `instabug.atlassian.net` as-is since it is the real Atlassian domain, and kept `Instabug/ios` since that repo does exist.

**Review by andrewdanieldev — APPROVED** [review]
